dhall-csv/src/Dhall/Csv.hs

```diff
@@ -29,38 +108,108 @@ import qualified Dhall.TypeCheck
 import qualified Dhall.Util
 import qualified System.FilePath
 
-data CompileError = Unsupported (Expr Void Void)
+{-| This is the exception type for errors that can arise when converting from
+    Dhall to CSV.
+
+    It contains information on the specific cases that might
+    fail to give a better insight.
+-}
+data CompileError
+    = Unsupported (Expr Void Void)
+    | NotAList (Expr Void Void)
+    | NotARecord (Expr Void Void)
+    | BareNone
 
 instance Show CompileError where
     show (Unsupported e) =
         Data.Text.unpack $
-            _ERROR <> ": Cannot translate to CSV                                             \n\
+            _ERROR <> ": Cannot translate record field to CSV                                \n\
+            \                                                                                \n\
+            \Explanation: Only the following types of record fields can be converted to CSV: \n\
+            \                                                                                \n\
+            \● ❰Bool❱                                                                        \n\
+            \● ❰Natural❱                                                                     \n\
+            \● ❰Integer❱                                                                     \n\
+            \● ❰Double❱                                                                      \n\
+            \● ❰Text❱                                                                        \n\
+            \● ❰Optional❱ tp (where tp is a valid record field type)                         \n\
+            \● Unions *                                                                      \n\
             \                                                                                \n\
-            \Explanation: Only records of primitive values can be                            \n\
-            \translated from Dhall to CSV.                                                   \n\
+            \* Unions can have empty alternatives or alternatives with valid                 \n\
+            \  record field types                                                            \n\
             \                                                                                \n\
             \The following Dhall expression could not be translated to CSV:                  \n\
             \                                                                                \n\
             \" <> insert e
```

Comment:
I have thought of that but couldn't think of a way to get the type (because all the `TypeCheck` functions return `Either` values). I just came up with a solution to use `fromRight` function and put the original expression as default (though this default should never be used, as the expression was already type checked). 

Would this solution be good?